obspy/io/rg16/core.py

```diff
@@ -0,0 +1,238 @@
+
+# geophone instrument code
+INSTRUMENT_CODE = 'P'
+
+# mapping for "standard_orientation"
+STANDARD_COMPONENT_MAP = {'2': 'Z', '3': 'N', '4': 'E'}
+
+# ------------------- read and format check functions
+
+
+@open_file
+def read_rg16(filename, headonly=False, starttime=None, endtime=None,
+              merge=False, contacts_north=False, **kwargs):
+    """
+    Read Fairfield Nodal's Receiver Gather File Format version 1.6-1.
+
+    :param filename: A path to the file or a buffer of an opened file.
+    :type filename: str, buffer
+    :param headonly: If True don't read data, only header information.
+    :type headonly: bool
+    :param starttime: If not None dont read traces that end before starttime.
+    :type starttime: optional, obspy.UTCDateTime
+    :param endtime: If None None dont read traces that start after endtime.
+    :type endtime: optional, obspy.UTCDateTime
+    :param merge:
+        If True merge contiguous data blocks as they are found. For
+        continuous data files having 100,000+ traces this will create
+        more manageable streams.
+    :type merge: bool
+    :param contacts_north:
+        Setting this parameter to True indicates the file either contains 1C
+        traces or that the instruments were deployed with the gold contact
+        terminals facing north. If this parameter is used, it will map the
+        components to Z, N, and E (if 3C) as well as correct the polarity for
+        the vertical component.
+    :type contacts_north: bool
+    :return: An ObsPy :class:`~obspy.core.stream.Stream` object.
+
+    """
+    if not is_rg16(filename):
+        raise ValueError('read_fcnt was not passed a Fairfield RG 1.6 file')
+    # get timestamps
+    time1 = UTCDateTime(starttime).timestamp if starttime else 0
+    time2 = UTCDateTime(endtime).timestamp if endtime else BIG_TS
+    # read general header information
+    gheader = _read_block(filename, general_header_block)
+    # byte number channel sets start at in file
+    chan_set_start = (gheader['num_additional_headers'] + 1) * 32
+    # get the byte number the extended headers start
+    eheader_start = (gheader['channel_sets']) * 32 + chan_set_start
+    # read trace headers
+    ex_headers = gheader['extended_headers'] + gheader['external_headers']
+    # get byte number trace headers start
+    theader_start = eheader_start + (ex_headers * 32)
+    # get traces and return stream
+    traces = _make_traces(filename, theader_start, gheader, head_only=headonly,
+                          starttime=time1, endtime=time2, merge=merge,
+                          standard_orientation=contacts_north)
+    return Stream(traces=traces)
+
+
+@open_file
+def is_rg16(filename, **kwargs):
+    """
+    Determine if a file or buffer contains an rg16 file.
+
+    :param filename: A path to the file or a buffer of an opened file.
+    :type filename: str, buffer
+    :return: bool
+    """
+    try:
+        filename.seek(0)
```

Comment:
I'd vote against seeking here as it would then enable passing a file-pointer anywhere in a file or byte stream. I'm not sure if this is useful for this particular format but at least it would be consistent with the rest of ObsPy.